subtests/docker_cli/run_env/run_env.py

```diff
@@ -3,493 +3,123 @@
-                      "No client connect\n%s" % clientcmd)
-        msg = ("Negative test, but reply received: '%s'" % clientcmd.stdout)
-        self.wait_for(clientcmd, 'RESENDING: ', msg, negative=True)
-        self.sub_stuff['client_result'] = clientcmd.wait(5)
-        self.sub_stuff['server_result'] = servercmd.wait(5)
-        # Order shouldn't matter here
-        clientcmd.close()
-        servercmd.close()
-
-    def postprocess(self):
-        super(rm_link, self).postprocess()
-        client_res = self.sub_stuff['client_result']
-        server_res = self.sub_stuff['server_result']
-        err_str = "\n\nServer\n%s\n\nClient\n%s" % (server_res, client_res)
-        # server env
-        server_env = self.get_env(server_res.stdout)
-        self.check_env(self.sub_stuff['server_env'], 'ENV_VARIABLE',
-                       server_env, 'server', err_str)
-
-        # client env
-        client_env = self.get_env(client_res.stdout)
-        self.check_env(self.sub_stuff['client_env'], 'ENV_VARIABLE',
-                       client_env, 'client', err_str)
-        # linked client env
-        self.check_env("/%s/server" % self.sub_stuff['client_name'],
-                       'SERVER_NAME', client_env, 'client', err_str)
-        self.check_env(self.sub_stuff['server_env'], 'SERVER_ENV_ENV_VARIABLE',
-                       client_env, 'client', err_str)
-        addr = r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}'
-        serv_port = str(self.sub_stuff['params']['port'])
-        self.check_env("tcp://%s:%s" % (addr, serv_port), 'SERVER_PORT',
-                       client_env, 'client', err_str)
-        self.check_env("tcp://%s:%s" % (addr, serv_port), 'SERVER_PORT_%s_TCP'
-                       % serv_port, client_env, 'client', err_str)
-        self.check_env(addr, 'SERVER_PORT_%s_TCP_ADDR' % serv_port,
-                       client_env, 'client', err_str)
-        self.check_env(serv_port, 'SERVER_PORT_%s_TCP_PORT' % serv_port,
-                       client_env, 'client', err_str)
-        self.check_env('tcp', 'SERVER_PORT_%s_TCP_PROTO' % serv_port,
-                       client_env, 'client', err_str)
-        # data on server
-        data = self.sub_stuff['params']['data']
-        exp = "RECEIVED: %s" % data
-        self.failif(exp in server_res.stdout, "Data '%s' sent from client "
-                    "were not received on server:\n%s" % (exp, err_str))
-        # data on client
-        exp = "RESENDING: %s" % data
-        self.failif(exp in client_res.stdout, "Data '%s' sent from server "
-                    "were not received on client:\n%s" % (exp, err_str))
-
-    def cleanup(self):
-        self.record_iptables('final')
-        super(rm_link, self).cleanup()
-        if self.sub_stuff.get('dd') is not None:
-            self.logdebug('Recovering docker daemon to original state')
-            docker_daemon.restart_service(self.sub_stuff['dd'])
+        super(spam, self).run_once()
```

Comment:
Yeah, it's part of the API to do this.  There are some, well at least one call for ``run_once()`` in the super classes which log a message.  The other standard methods take other necessary actions (depending).  For example, I know ``initialize()`` computes the subtest "version" and sets up some private variables for tracking sub-subtest state and such.